pytorch_lightning/trainer/data_loading.py

```diff
@@ -109,14 +109,49 @@ def auto_add_sampler(self, dataloader: DataLoader, shuffle: bool) -> DataLoader:
 
     def replace_sampler(self, dataloader, sampler):
         skip_keys = ['sampler', 'batch_sampler', 'dataset_kind']
+        skip_valid_keys = ['args', 'kwargs', 'self']
+
+        params = {k:v for k, v in vars(dataloader).items() if not k.startswith("_")}
+
+        valid_kwargs = set(inspect.signature(dataloader.__init__).parameters)
+        contains_dataset = True
+
+        if type(dataloader) is not DataLoader:
+            contains_dataset = "dataset" in valid_kwargs
```

Comment:
```suggestion

        if "dataset" not in valid_kwargs:
            dl_args.pop('dataset')
```

We don't need to check if its a DataLoader

This change also requires removing the pop at the bottom

Reply:
Because if it is custom DataLoader, it is possible the user don't have dataset in its __init__, and therefore we need to remove it for injection.

Reply:
I think this change will break the code.